2996_reflection/reflection.md

```diff
@@ -1295,19 +1301,25 @@ The reflection operator produces a reflection value from a grammatical construct
 > |       `^` `::`
 > |       `^` _namespace-name_
 > |       `^` _type-id_
-> |       `^` _cast-expression_
+> |       `^` _id-expression_
+
+The expression `^::` evaluates to a reflection of the global namespace. When the operand is a _namespace-name_ or _type-id_, the resulting value is a reflection of the designated namespace or type.
+
+When the operand is an _id-expression_, the resulting value is a reflection of the designated entity found by lookup. This might be any of:
+
+- a variable, static data member, or structured binding
+- a function or member function
+- a nonstatic data member
+- a template or member template
+- an enumerator
 
-Note that _cast-expression_ includes _id-expression_, which in turn can designate templates, member names, etc.
+For all other operands, the expression is ill-formed. In a SFINAE context, a failure to substitute the operand of a reflection operator construct causes that construct to not evaluate to constant.
 
-The current proposal requires that the _cast-expression_ be:
+Earlier revisions of this paper allowed for taking the reflection of any _cast-expression_ that could be evaluated as a constant expression, as we believed that a constant expression could be internally "represented" by just capturing the value to which it evaluated. However, the possibility of side effects from constant evaluation (introduced by this very paper) renders this approach infeasible: even a constant expression would have to be evaluated every time it's spliced. It was ultimately decided to defer all support for expression reflection to a future paper.
```

Comment:
I think we should note that we intend the syntax for that to be `^(expression)` so that it's clear we're looking ahead?

Reply:
Done.